sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/QueryStageExec.scala

```diff
@@ -154,14 +154,14 @@ abstract class QueryStageExec extends LeafExecNode {
 abstract class ExchangeQueryStageExec extends QueryStageExec {
 
   /**
-   * Cancel the stage materialization if in progress; otherwise do nothing.
+   * Cancel the stage materialization if in progress with an optional reason; otherwise do nothing.
    */
-  final def cancel(): Unit = {
+  final def cancel(reason: Option[String]): Unit = {
```

Comment:
does any caller pass `None`?

Reply:
I don't think so, are you suggesting we can change the method signature to `reason: String` then?